### PR TITLE
Refactor tvl on backend

### PR DESCRIPTION
**Pull request by torztomasz**
Resolves L2B-2817

## Important ENV change
`ERROR_ON_UNSYNCED_DETAILED_TVL` -> `ERROR_ON_UNSYNCED_TVL`

### <samp>🤖 Generated by Copilot at 9f477a9</samp>

This pull request refactors the TVL API to remove the distinction between detailed and aggregated TVL data, and to add new features such as grouping, filtering, and breakdown of assets by value type. It renames and updates several types, functions, modules, and files to reflect the new TVL API structure and logic. It also removes unused code and simplifies the configuration and testing of the TVL API.

### <samp>🤖 Generated by Copilot at 9f477a9</samp>

*  Refactor TVL API to support new features and endpoints ([link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f9513ea8eba4de121ba87a982d844520decf2b7ad369eaee095360ba4361a7a1L2-R16),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f9513ea8eba4de121ba87a982d844520decf2b7ad369eaee095360ba4361a7a1L15-R36),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f9513ea8eba4de121ba87a982d844520decf2b7ad369eaee095360ba4361a7a1L26-R78),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f9513ea8eba4de121ba87a982d844520decf2b7ad369eaee095360ba4361a7a1L66-R135),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f9513ea8eba4de121ba87a982d844520decf2b7ad369eaee095360ba4361a7a1L92-R213),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f9513ea8eba4de121ba87a982d844520decf2b7ad369eaee095360ba4361a7a1L121-R260),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f9513ea8eba4de121ba87a982d844520decf2b7ad369eaee095360ba4361a7a1L159-R335),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f9513ea8eba4de121ba87a982d844520decf2b7ad369eaee095360ba4361a7a1R357-R397),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-0ca1e217a9ce66930b1e1edaf7019c02464d5d51b0504c116a3411acd1d6865cL4-R10),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-0ca1e217a9ce66930b1e1edaf7019c02464d5d51b0504c116a3411acd1d6865cL17-R17),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-0ca1e217a9ce66930b1e1edaf7019c02464d5d51b0504c116a3411acd1d6865cL37-R39),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-0ca1e217a9ce66930b1e1edaf7019c02464d5d51b0504c116a3411acd1d6865cL46-R46),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f2dd6408569eff76c069b15b4238974df72ce95506b223fdcabf6699c62fdb10L2-R2),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f2dd6408569eff76c069b15b4238974df72ce95506b223fdcabf6699c62fdb10L10-R10),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f2dd6408569eff76c069b15b4238974df72ce95506b223fdcabf6699c62fdb10L22-R25),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f2dd6408569eff76c069b15b4238974df72ce95506b223fdcabf6699c62fdb10L32-R32),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f2dd6408569eff76c069b15b4238974df72ce95506b223fdcabf6699c62fdb10L49-R52),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f2dd6408569eff76c069b15b4238974df72ce95506b223fdcabf6699c62fdb10L59-R59),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f2dd6408569eff76c069b15b4238974df72ce95506b223fdcabf6699c62fdb10L102-R120),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f2dd6408569eff76c069b15b4238974df72ce95506b223fdcabf6699c62fdb10L158-R161),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-6cda84458bd352d3e5ff3c17c90d3111795fc67635ec89920fc56a850a3b66dcR2-R3),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-6cda84458bd352d3e5ff3c17c90d3111795fc67635ec89920fc56a850a3b66dcL7-R178),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-6cda84458bd352d3e5ff3c17c90d3111795fc67635ec89920fc56a850a3b66dcL66-R250),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-d6e7078789d03171efe81071af349f0ecd299179cb480754f80e3707dd369a5dL11-R13),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-8f56172b6b95395bb17dc743603bd06725200cf75f30c3f31165a840f0265661L6),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-8f56172b6b95395bb17dc743603bd06725200cf75f30c3f31165a840f0265661R11),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-8f56172b6b95395bb17dc743603bd06725200cf75f30c3f31165a840f0265661L52-R56),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-808e76957870c912473504ffc33c12df11bc9044a8b92f40e34cb30b1b087012L11-R34),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-808e76957870c912473504ffc33c12df11bc9044a8b92f40e34cb30b1b087012L34-R48),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-808e76957870c912473504ffc33c12df11bc9044a8b92f40e34cb30b1b087012L57-R71),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-808e76957870c912473504ffc33c12df11bc9044a8b92f40e34cb30b1b087012L139-R98),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-808e76957870c912473504ffc33c12df11bc9044a8b92f40e34cb30b1b087012L159-R105),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-808e76957870c912473504ffc33c12df11bc9044a8b92f40e34cb30b1b087012L166-R112),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-e3732d4542f41c46af007d90e6cbc4720c8730cb9c883577ac049ca60ebcd73aL54-R54),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-421689afaba0ad2e3d74a975dbc16a85149ef8feb352e8f57935d0d9b94460c2L13-R13),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-421689afaba0ad2e3d74a975dbc16a85149ef8feb352e8f57935d0d9b94460c2L68-R65),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-854c01b262be6e7cb7b7b373ff4077fc64b1eb9c8b62d1d4e12ff0f1f66c8aa0L15-R15),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-854c01b262be6e7cb7b7b373ff4077fc64b1eb9c8b62d1d4e12ff0f1f66c8aa0L75-R72),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-084e015386d9347e95ef97547c549e1041208d1cbe32be127ba0b4f08d8b04f6L6),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-084e015386d9347e95ef97547c549e1041208d1cbe32be127ba0b4f08d8b04f6L138-R139),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-084e015386d9347e95ef97547c549e1041208d1cbe32be127ba0b4f08d8b04f6L159-R148),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-084e015386d9347e95ef97547c549e1041208d1cbe32be127ba0b4f08d8b04f6L165-R154),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-ca28803a64272e59151496fdc3ed8b64bfb9de6bf3aa620c25e29f7d10733226L87-L187),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-8e940c1173227ab0c6acb6a42b87d930b621b117002e96fb5bf9c9654a5e5df6L96-R97),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-8e940c1173227ab0c6acb6a42b87d930b621b117002e96fb5bf9c9654a5e5df6L171-R116),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-8e940c1173227ab0c6acb6a42b87d930b621b117002e96fb5bf9c9654a5e5df6L192-R137),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-025853d2b2edb6af6003d1eaa0299bd54ad5ecbeea5fe01b2e30fc06b7272e7cL4-R5),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-025853d2b2edb6af6003d1eaa0299bd54ad5ecbeea5fe01b2e30fc06b7272e7cL17-R17),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-025853d2b2edb6af6003d1eaa0299bd54ad5ecbeea5fe01b2e30fc06b7272e7cL35),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-025853d2b2edb6af6003d1eaa0299bd54ad5ecbeea5fe01b2e30fc06b7272e7cL55-R48),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-025853d2b2edb6af6003d1eaa0299bd54ad5ecbeea5fe01b2e30fc06b7272e7cL66-L120),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-a23f78a2e2544c5cfc7a62289f58b1f2616eac23fa93f30358951b1cded47a9c))
  * Rename and update functions and types to avoid confusion and reflect new logic ([link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-0ca1e217a9ce66930b1e1edaf7019c02464d5d51b0504c116a3411acd1d6865cL4-R10),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-0ca1e217a9ce66930b1e1edaf7019c02464d5d51b0504c116a3411acd1d6865cL17-R17),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-0ca1e217a9ce66930b1e1edaf7019c02464d5d51b0504c116a3411acd1d6865cL37-R39),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-0ca1e217a9ce66930b1e1edaf7019c02464d5d51b0504c116a3411acd1d6865cL46-R46),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f2dd6408569eff76c069b15b4238974df72ce95506b223fdcabf6699c62fdb10L2-R2),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f2dd6408569eff76c069b15b4238974df72ce95506b223fdcabf6699c62fdb10L10-R10),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f2dd6408569eff76c069b15b4238974df72ce95506b223fdcabf6699c62fdb10L22-R25),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f2dd6408569eff76c069b15b4238974df72ce95506b223fdcabf6699c62fdb10L32-R32),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f2dd6408569eff76c069b15b4238974df72ce95506b223fdcabf6699c62fdb10L49-R52),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f2dd6408569eff76c069b15b4238974df72ce95506b223fdcabf6699c62fdb10L59-R59),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f2dd6408569eff76c069b15b4238974df72ce95506b223fdcabf6699c62fdb10L102-R120),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f2dd6408569eff76c069b15b4238974df72ce95506b223fdcabf6699c62fdb10L158-R161),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-6cda84458bd352d3e5ff3c17c90d3111795fc67635ec89920fc56a850a3b66dcL66-R250),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-8f56172b6b95395bb17dc743603bd06725200cf75f30c3f31165a840f0265661L52-R56),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-808e76957870c912473504ffc33c12df11bc9044a8b92f40e34cb30b1b087012L34-R48),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-808e76957870c912473504ffc33c12df11bc9044a8b92f40e34cb30b1b087012L57-R71),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-808e76957870c912473504ffc33c12df11bc9044a8b92f40e34cb30b1b087012L139-R98),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-808e76957870c912473504ffc33c12df11bc9044a8b92f40e34cb30b1b087012L166-R112),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-e3732d4542f41c46af007d90e6cbc4720c8730cb9c883577ac049ca60ebcd73aL54-R54),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-421689afaba0ad2e3d74a975dbc16a85149ef8feb352e8f57935d0d9b94460c2L13-R13),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-421689afaba0ad2e3d74a975dbc16a85149ef8feb352e8f57935d0d9b94460c2L68-R65),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-854c01b262be6e7cb7b7b373ff4077fc64b1eb9c8b62d1d4e12ff0f1f66c8aa0L15-R15),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-854c01b262be6e7cb7b7b373ff4077fc64b1eb9c8b62d1d4e12ff0f1f66c8aa0L75-R72),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-8e940c1173227ab0c6acb6a42b87d930b621b117002e96fb5bf9c9654a5e5df6L171-R116),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-8e940c1173227ab0c6acb6a42b87d930b621b117002e96fb5bf9c9654a5e5df6L192-R137),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-025853d2b2edb6af6003d1eaa0299bd54ad5ecbeea5fe01b2e30fc06b7272e7cL17-R17),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-025853d2b2edb6af6003d1eaa0299bd54ad5ecbeea5fe01b2e30fc06b7272e7cL55-R48))
  * Add new functions and types to handle new endpoints and aggregation logic ([link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f9513ea8eba4de121ba87a982d844520decf2b7ad369eaee095360ba4361a7a1L2-R16),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f9513ea8eba4de121ba87a982d844520decf2b7ad369eaee095360ba4361a7a1L15-R36),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f9513ea8eba4de121ba87a982d844520decf2b7ad369eaee095360ba4361a7a1L26-R78),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f9513ea8eba4de121ba87a982d844520decf2b7ad369eaee095360ba4361a7a1L92-R213),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f9513ea8eba4de121ba87a982d844520decf2b7ad369eaee095360ba4361a7a1L121-R260),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f9513ea8eba4de121ba87a982d844520decf2b7ad369eaee095360ba4361a7a1L159-R335),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-6cda84458bd352d3e5ff3c17c90d3111795fc67635ec89920fc56a850a3b66dcR2-R3),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-6cda84458bd352d3e5ff3c17c90d3111795fc67635ec89920fc56a850a3b66dcL66-R250),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-8f56172b6b95395bb17dc743603bd06725200cf75f30c3f31165a840f0265661R11),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-808e76957870c912473504ffc33c12df11bc9044a8b92f40e34cb30b1b087012L11-R34),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-808e76957870c912473504ffc33c12df11bc9044a8b92f40e34cb30b1b087012L159-R105))
  * Update existing functions and types to use new report types and grouping structures from `tvl` module ([link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f9513ea8eba4de121ba87a982d844520decf2b7ad369eaee095360ba4361a7a1L66-R135),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-6cda84458bd352d3e5ff3c17c90d3111795fc67635ec89920fc56a850a3b66dcL66-R250),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-808e76957870c912473504ffc33c12df11bc9044a8b92f40e34cb30b1b087012L57-R71))
  * Move `getDataTimings` function to a separate file and reuse it in other controllers ([link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-f9513ea8eba4de121ba87a982d844520decf2b7ad369eaee095360ba4361a7a1R357-R397))
  * Delete `DetailedTvlController` class and related functions and types, as they are no longer needed ([link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-a23f78a2e2544c5cfc7a62289f58b1f2616eac23fa93f30358951b1cded47a9c),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-d6e7078789d03171efe81071af349f0ecd299179cb480754f80e3707dd369a5dL11-R13),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-8f56172b6b95395bb17dc743603bd06725200cf75f30c3f31165a840f0265661L6),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-084e015386d9347e95ef97547c549e1041208d1cbe32be127ba0b4f08d8b04f6L6),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-084e015386d9347e95ef97547c549e1041208d1cbe32be127ba0b4f08d8b04f6L138-R139),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-084e015386d9347e95ef97547c549e1041208d1cbe32be127ba0b4f08d8b04f6L159-R148),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-084e015386d9347e95ef97547c549e1041208d1cbe32be127ba0b4f08d8b04f6L165-R154),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-ca28803a64272e59151496fdc3ed8b64bfb9de6bf3aa620c25e29f7d10733226L87-L187),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-8e940c1173227ab0c6acb6a42b87d930b621b117002e96fb5bf9c9654a5e5df6L96-R97),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-025853d2b2edb6af6003d1eaa0299bd54ad5ecbeea5fe01b2e30fc06b7272e7cL4-R5),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-025853d2b2edb6af6003d1eaa0299bd54ad5ecbeea5fe01b2e30fc06b7272e7cL35),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-025853d2b2edb6af6003d1eaa0299bd54ad5ecbeea5fe01b2e30fc06b7272e7cL66-L120))
  * Delete placeholder API data, as it is no longer needed ([link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-025853d2b2edb6af6003d1eaa0299bd54ad5ecbeea5fe01b2e30fc06b7272e7cL35),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-025853d2b2edb6af6003d1eaa0299bd54ad5ecbeea5fe01b2e30fc06b7272e7cL66-L120))
  * Move `createTvlRouter` function from `TvlRouter` module to `TvlController` module ([link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-6cda84458bd352d3e5ff3c17c90d3111795fc67635ec89920fc56a850a3b66dcL7-R178),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-808e76957870c912473504ffc33c12df11bc9044a8b92f40e34cb30b1b087012L11-R34),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-084e015386d9347e95ef97547c549e1041208d1cbe32be127ba0b4f08d8b04f6L138-R139),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-084e015386d9347e95ef97547c549e1041208d1cbe32be127ba0b4f08d8b04f6L159-R148))
  * Rename `generateTvlApiResponse.ts` file to `TvlController.ts`, as it now contains more functions related to the TVL API controller logic ([link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-6cda84458bd352d3e5ff3c17c90d3111795fc67635ec89920fc56a850a3b66dcL7-R178))
  * Rename `detailedTvl.test.ts` file to `tvl.test.ts`, as it now contains tests for the new tvl module functions ([link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-d6e7078789d03171efe81071af349f0ecd299179cb480754f80e3707dd369a5dL11-R13))
  * Move `router.get('/api/tvl')` route to the top of the `createTvlRouter` function, as it is the main endpoint for the TVL API ([link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-808e76957870c912473504ffc33c12df11bc9044a8b92f40e34cb30b1b087012L57-R71))
  * Update variable names to match new function names and return types ([link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-808e76957870c912473504ffc33c12df11bc9044a8b92f40e34cb30b1b087012L159-R105),[link](https://github.com/l2beat/l2beat/pull/2068/files?diff=unified&w=0#diff-025853d2b2edb6af6003d1eaa0299bd54ad5ecbeea5fe01b2e30fc06b7272e7cL55-R48))

### CODE REVIEW

File: packages/backend/src/config/config.local.ts [hunk 1]
```diff
@@ -10,10 +10,7 @@ import { getGitCommitHash } from './getGitCommitHash'
 
 export function getLocalConfig(env: Env): Config {
   const tvlEnabled = env.boolean('TVL_ENABLED', true)
-  const errorOnUnsyncedDetailedTvl = env.boolean(
-    'ERROR_ON_UNSYNCED_DETAILED_TVL',
-    false,
-  )
+  const errorOnUnsyncedTvl = env.boolean('ERROR_ON_UNSYNCED_TVL', false)
```
Comment (antooni) [inline packages/backend/src/config/config.local.ts:13]: please update backend readme

File: packages/frontend/src/build/api/fetchTvlApi.ts [hunk 2]
```diff
@@ -16,55 +19,111 @@ export async function fetchTvlApi(
   if (backend.mock) {
     return getMockTvlApiResponse()
   }
-  const url = backend.apiUrl + '/api/tvl'
+  const url = `${backend.apiUrl}/api/detailed-tvl`
```
Comment (antooni) [inline packages/frontend/src/build/api/fetchTvlApi.ts:22]:
```suggestion
  const url = `${backend.apiUrl}/api/tvl`
```

**Review by antooni — COMMENTED** [review]
There are still some "detailed leftovers":
- backend/README.md
- frontend/src/build/buildMetaImages.ts
- frontend/src/build/api/fetchTvlApi.ts
- frontend/src/components/chart/RadioChartTypeControl.tsx
- frontend/src/components/navigation-tabs/ScalingNavigationTabs.tsx
- frontend/src/components/table/props/getScalingRowProps.ts
- frontend/src/components/table/props/getScalingTableColumns.tsx
- frontend/src/pages/index.ts
- frontend/src/pages/meta-images/DetailedTvlMetaImage.tsx
- frontend/src/pages/meta-images/getProps.ts
- frontend/src/pages/meta-images/index.tsx
- frontend/src/pages/scaling/types.ts
- frontend/src/pages/scaling/detailed-tvl/index.tsx
- frontend/src/pages/scaling/detailed-tvl/types.ts
- frontend/src/pages/scaling/detailed-tvl/props/getPageMetadata.ts
- frontend/src/pages/scaling/detailed-tvl/props/getProps.ts
- frontend/src/pages/scaling/detailed-tvl/props/getScalingDetailedTvlView.ts
- frontend/src/pages/scaling/detailed-tvl/view/ScalingDetailedTvlPage.tsx
- frontend/src/pages/scaling/detailed-tvl/view/ScalingDetailedTvlPage.tsx
- frontend/src/pages/scaling/detailed-tvl/view/ScalingDetailedTvlView.tsx
- frontend/src/pages/scaling/projects/props/getProjectHeader.ts
- frontend/src/pages/scaling/projects-tvl-breakdown/props/getTvlBreakdownView.ts
- frontend/src/scripts/charts/ChartControls.ts
- frontend/src/scripts/charts/types.ts
- frontend/src/scripts/charts/data-controller/ChartDataController.ts
- frontend/src/scripts/charts/view-controller/ChartViewController.ts
- frontend/src/scripts/charts/view-controller/hovers.ts
- frontend/src/scripts/charts/view-controller/types.ts
- frontend/src/scripts/charts/view-controller/charts/getDetailedTvlRenderParams.ts
- frontend/src/utils/project/getChart.ts
- frontend/src/utils/tvl/getTvlWithChange.ts

**Comment by torztomasz** [comment]
The distinction between detailed TVL and TVL is needed on frontend side.

**Comment by antooni** [comment]
tbh I would get rid of the detailed everywhere, and treat the detailed as new TVL, for the frontend maybe we can introduce a "summary" distinction

File: packages/frontend/src/build/api/fetchTvlApi.ts [hunk 3]
```diff
@@ -3,37 +3,37 @@ import {
...
   backend: Config['backend'],
   http: JsonHttpClient,
-): Promise<DetailedTvlApiResponse> {
+): Promise<TvlApiResponse> {
   if (backend.mock) {
-    return getMockDetailedTvlApiResponse()
+    return getMockTvlApiResponse()
   }
-  const url = `${backend.apiUrl}/api/detailed-tvl`
+  const url = `${backend.apiUrl}/api/-tvl`
```
Comment (antooni) [inline packages/frontend/src/build/api/fetchTvlApi.ts:22]:
```suggestion
  const url = `${backend.apiUrl}/api/tvl`
```

File: yarn.lock [hunk 4]
```diff
@@ -1966,10 +1966,10 @@
   dependencies:
     zod "^3.22.2"
 
-"@l2beat/discovery@0.18.4":
-  version "0.18.4"
-  resolved "https://registry.yarnpkg.com/@l2beat/discovery/-/discovery-0.18.4.tgz#00ec16187f653fd548cbd22442309b1ec5a61265"
-  integrity sha512-w/PtXkJRwrvAtTJgL0FvKteXZ52OeBB8HSGqqSt3rR2fYg0LWawDeq2bao8ofR8dfDEI63IJhhxvBfsjep912Q==
+"@l2beat/discovery@0.21.0":
```
Comment (antooni) [inline yarn.lock:1969]: why?

**Review by antooni — APPROVED** [review]

**Pull request merged**